src/datasets/builder.py

```diff
@@ -507,7 +507,7 @@ def download_and_prepare(
         with FileLock(lock_path):
             data_exists = os.path.exists(self._cache_dir)
             if data_exists and download_mode == GenerateMode.REUSE_DATASET_IF_EXISTS:
-                logger.warning("Reusing dataset %s (%s)", self.name, self._cache_dir)
+                logger.info("Reusing dataset %s (%s)", self.name, self._cache_dir)
```

Comment:
I think this should stay a warning.
When something is reloaded from the cache we should show it to the users or it could lead to very bad experiences. For example when users are doing changes and expect things to get recomputed instead of reloaded.

Especially since caching is not something that intuitive for some users: what causes recomputing vs what allows things to be reloaded is not something you can guess easily without reading the documentation.

Reply:
tl;dr: given that the default level is WARNING and this is important then yes, I totally agree.

-------------

the longer view.

I totally hear that this is very important.

I think the key here is to discuss what is expected in the normal usage, when the person fires off the same code again and again vs develop/tune up/troubleshooting.

So to me a warning is something that is bordering-line to errors, and I only want to see such notifications when something could be going wrong with the results.

If we agree than an average user wants to enable INFO (as we are doing now for when transformers examples are used) then it's already included. For example, to stress the significance the wording could be changed to:

```
logger.info("Important: Reusing dataset %s (%s)", self.name, self._cache_dir)
```

If most of the time users run `datasets` with verbosity `WARNING` then indeed it will be missed and then you're absolutely correct, it should remain under `logger.warning`.

In other words they way I think an efficient development cycle is:

- normally run with the highest log-level and keep things silent
- when something doesn't work as expected or I'm filing a bug report - run with INFO or DEBUG enabled and inspect/share the logs.

So in this case I say to myself: `datasets` warnings are really `info`s so I'll switch to ERROR to remove the noise.

Frankly, `transformers` has many warnings which are in the same boat - not really important most of the time. So I've switched to `--log_level error` to compensate for that noise, and it's awesome! (and of course we had the env vars all along, but these are more difficult to use)

Reply:
Hmm, but it logs this on every instance, e.g. currently with 2 processes:
```
06/24/2021 19:01:54 - WARNING - datasets.arrow_dataset -   Loading cached processed dataset at /home/stas/.cache/huggingface/datasets/wmt16/ro-en/1.0.0/0d9fb3e814712c785176ad8cdb9f465fbe6479000ee6546725db30ad8a8b5f8a/cache-1bb178ab5058b59a.arrow
06/24/2021 19:01:54 - WARNING - datasets.arrow_dataset -   Loading cached processed dataset at /home/stas/.cache/huggingface/datasets/wmt16/ro-en/1.0.0/0d9fb3e814712c785176ad8cdb9f465fbe6479000ee6546725db30ad8a8b5f8a/cache-1bb178ab5058b59a.arrow
```
So if I have 64 processes this will be a lot of identical reports.

But I guess `datasets` can't be aware of ranks and distributed environment :(

OK, the user fixes this issue with `--log_on_each_node 0 --log_level_replica error` which is already part of the HF Trainer and soon in all examples. And voila now there is only one warning no matter how many nodes and gpus there are.

All is good!

Reply:
I've reverted this change.